File: demo/hello.c
#include <signal.h>
#include <stdio.h>
#include <unistd.h>

int main() {
    pid_t pid = getpid();
    printf("Hello: pid %d\n", pid);
    pid_t ppid = getppid();
    printf("Hello: parent pid %d\n", ppid);
    /* sleep(3);
    printf("\nHello: sending SIGUSR1 to parent %d \n", ppid);
    kill(ppid, SIGUSR1); */
    printf("Hello World \n");
}

File: demo/Hello.java
import java.io.IOException;
import java.lang.management.ManagementFactory;

public class Hello {
    public static void main(String[] args) {
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            System.out.println("HelloWorld interrupted");
        }
        long pid = ManagementFactory.getRuntimeMXBean().getPid();
        String p = ManagementFactory.getRuntimeMXBean().getSystemProperties().get("PID");
        System.out.println("PID: " + pid + " " + p);
        try {
            Runtime.getRuntime().exec("kill -SIGUSER1 " + pid);
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.out.println("Hello, World");
    }
}
